test/e2e_node/configmap.go

```diff
@@ -0,0 +1,371 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e_node
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/util"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = framework.KubeDescribe("ConfigMap", func() {
+
+	f := NewDefaultFramework("configmap")
+
+	It("should be consumable from pods in volumpe [Conformance]", func() {
+		doConfigMapE2EWithoutMappings(f, 0, 0)
+	})
+
+	It("should be consumable from pods in volume as non-root [Conformance]", func() {
+		doConfigMapE2EWithoutMappings(f, 1000, 0)
+	})
+
+	It("should be consumable from pods in volume as non-root with FSGroup [Feature:FSGroup]", func() {
+		doConfigMapE2EWithoutMappings(f, 1000, 1001)
+	})
+
+	It("should be consumable from pods in volume with mappings [Conformance]", func() {
+		doConfigMapE2EWithMappings(f, 0, 0)
+	})
+
+	It("should be consumable from pods in volume with mappings as non-root [Conformance]", func() {
+		doConfigMapE2EWithMappings(f, 1000, 0)
+	})
+
+	It("should be consumable from pods in volume with mappings as non-root with FSGroup [Feature:FSGroup]", func() {
+		doConfigMapE2EWithMappings(f, 1000, 1001)
+	})
+
+	It("updates should be reflected in volume [Conformance]", func() {
+
+		// We may have to wait or a full sync period to elapse before the
+		// Kubelet projects the update into the volume and the container picks
+		// it up. This timeout is based on the default Kubelet sync period (1
+		// minute) plus additional time for fudge factor.
+		const podLogTimeout = 300 * time.Second
+
+		name := "configmap-test-upd-" + string(util.NewUUID())
+		volumeName := "configmap-volume"
+		volumeMountPath := "/etc/configmap-volume"
+		containerName := "configmap-volume-test"
+
+		configMap := &api.ConfigMap{
+			ObjectMeta: api.ObjectMeta{
+				Namespace: f.Namespace.Name,
+				Name:      name,
+			},
+			Data: map[string]string{
+				"data-1": "value-1",
+			},
+		}
+
+		By(fmt.Sprintf("Creating configMap with name %s", configMap.Name))
+		defer func() {
```

Comment:
Don't need to delete the configmap manually. The framework will delete the entire namespace after the test completes.